### PR TITLE
[editor] HuggingFaceAutomaticSpeechRecognitionPromptSchema

### PR DESCRIPTION
**Pull request by rholinshead**
[editor] HuggingFaceAutomaticSpeechRecognitionPromptSchema

---
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/lastmile-ai/aiconfig/pull/818).
* __->__ #818
* #815
* #814

### CODE REVIEW

File: python/src/aiconfig/editor/client/src/shared/prompt_schemas/HuggingFaceAutomaticSpeechRecognitionPromptSchema.ts
Comment (saqadri) [inline python/src/aiconfig/editor/client/src/shared/prompt_schemas/HuggingFaceAutomaticSpeechRecognitionPromptSchema.ts:8]: Same question as in previous diff for this

File: python/src/aiconfig/editor/client/src/shared/prompt_schemas/HuggingFaceAutomaticSpeechRecognitionPromptSchema.ts
Comment (saqadri) [inline python/src/aiconfig/editor/client/src/shared/prompt_schemas/HuggingFaceAutomaticSpeechRecognitionPromptSchema.ts:36]: input length for... in?

**Review by saqadri — APPROVED** [review]
Ship it! We can test after the model parsers land

File: python/src/aiconfig/editor/client/src/shared/prompt_schemas/HuggingFaceAutomaticSpeechRecognitionPromptSchema.ts
Comment (rossdanlm) [inline python/src/aiconfig/editor/client/src/shared/prompt_schemas/HuggingFaceAutomaticSpeechRecognitionPromptSchema.ts:21]: cc @Ankush-lastmile what formats does the ASR model parser support as inputs?

File: python/src/aiconfig/editor/client/src/shared/prompt_schemas/HuggingFaceAutomaticSpeechRecognitionPromptSchema.ts
Comment (rossdanlm) [reply python/src/aiconfig/editor/client/src/shared/prompt_schemas/HuggingFaceAutomaticSpeechRecognitionPromptSchema.ts:8]: Another thing to note is that the completion params aren't always included in description. Ex: `binary_output` is in the completion params but not explained
Screenshot 2024-01-09 at 12 49 41

**Review by rossdanlm — APPROVED** [review]
lgtm, sync with Ankush on the input format for audio, but we can also just land and fix forward

File: python/src/aiconfig/editor/client/src/shared/prompt_schemas/HuggingFaceAutomaticSpeechRecognitionPromptSchema.ts
Comment (rossdanlm) [inline python/src/aiconfig/editor/client/src/shared/prompt_schemas/HuggingFaceAutomaticSpeechRecognitionPromptSchema.ts:55]: Let's also include `max_new_tokens`: https://huggingface.co/docs/transformers/v4.36.1/en/main_classes/pipelines#transformers.AutomaticSpeechRecognitionPipeline.__call__

Comment (rossdanlm) [reply python/src/aiconfig/editor/client/src/shared/prompt_schemas/HuggingFaceAutomaticSpeechRecognitionPromptSchema.ts:55]: They also seem to use the same generation params as from text generation: https://huggingface.co/docs/transformers/en/main_classes/text_generation#transformers.GenerationConfig

I think it would be helpful if we could centralize this and have one schema inherit from another, but not sure how simple that is. Worse case we parse the JSON and do `dict.update()` to the `model_settings` field. if not possible immediately, we can create task for it

Comment (rholinshead) [reply python/src/aiconfig/editor/client/src/shared/prompt_schemas/HuggingFaceAutomaticSpeechRecognitionPromptSchema.ts:55]: We can easily just create a common prompt schema and spread it into the object then override what's needed

**Comment by rholinshead** [comment]
Just going to land current stack for now, and we can update on top. I just want a fresh stack to do some minor refactoring to prep for packaging. In light of that, I will transition off prompt schema work to work on the packaging and transition back in a bit

We can address comments on top as ASR parser is implemented and make sure the schema matches once that's done.

For defaults -- see comment in previous PR of why I'd prefer not to set defaults for now

File: python/src/aiconfig/editor/client/src/shared/prompt_schemas/HuggingFaceAutomaticSpeechRecognitionPromptSchema.ts
Comment (rossdanlm) [reply python/src/aiconfig/editor/client/src/shared/prompt_schemas/HuggingFaceAutomaticSpeechRecognitionPromptSchema.ts:55]: Followup issue to track: https://github.com/lastmile-ai/aiconfig/issues/836

**Pull request merged**
